src/near2far.cpp

```diff
@@ -331,6 +332,19 @@ realnum *dft_near2far::get_farfields_array(const volume &where, int &rank, size_
     }
   }
   sum_to_all(EH_, EH, 6 * 2 * N * Nfreq);
+
+  /* collapse singleton dimensions */
+  const int ndims = 4;
+  size_t collapsed_dims[ndims] = {};
+  int collapsed_rank = 0;
+  for (int i = 0; i < rank; ++i) {
+    if (dims[i] > 1) {
+      collapsed_dims[collapsed_rank++] = dims[i];
+    }
+  }
+  rank = collapsed_rank;
+  std::memcpy(dims, collapsed_dims, ndims * sizeof(size_t));
```

Comment:
Seems like it would be cleaner to do:
```cxx
/* collapse singleton dimensions */
int ireduced = 0;
for (int i = 0; i < rank; ++i)
    if (dims[i] > 1) dims[ireduced++] = dims[i];
rank = ireduced;
```

Reply:
Done.